plugin/src/main/java/org/opensearch/ml/action/model_group/DeleteModelGroupTransportAction.java

```diff
@@ -84,26 +83,14 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Delete
                     SearchRequest searchRequest = new SearchRequest(ML_MODEL_INDEX).source(searchSourceBuilder);
                     client.search(searchRequest, ActionListener.wrap(mlModels -> {
                         if (mlModels == null || mlModels.getHits().getTotalHits() == null || mlModels.getHits().getTotalHits().value == 0) {
-                            client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
-                                @Override
-                                public void onResponse(DeleteResponse deleteResponse) {
-                                    log.debug("Completed Delete Model Group Request, task id:{} deleted", modelGroupId);
-                                    wrappedListener.onResponse(deleteResponse);
-                                }
-
-                                @Override
-                                public void onFailure(Exception e) {
-                                    log.error("Failed to delete ML Model Group " + modelGroupId, e);
-                                    wrappedListener.onFailure(e);
-                                }
-                            });
+                            deleteModelGroup(deleteRequest, modelGroupId, wrappedListener);
                         } else {
                             throw new MLValidationException("Cannot delete the model group when it has associated model versions");
                         }
 
                     }, e -> {
                         if (e instanceof IndexNotFoundException) {
-                            wrappedListener.onFailure(new MLResourceNotFoundException("Fail to find model group"));
+                            deleteModelGroup(deleteRequest, modelGroupId, wrappedListener);
```

Comment:
When index is not found why are we trying to delete the model again?

Reply:
The index not found is the model version index. So here before deleting a model group, we are searching model index to check if there are any associated model versions in which case we do not delete the model group. But if index itself does not exit, it means there are no model versions indeed on the cluster and we can go ahead with deleting the model group